pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -503,6 +519,16 @@ func (m *manager) reconcileState() (success []reconciledContainer, failure []rec
 	return success, failure
 }
 
+func StaticCPUPolicyConditionsSatisfied(cpuManagerPolicyName string, podQos v1.PodQOSClass, cpuRequest *apiresource.Quantity) bool {
+	// returns true if
+	// 1. cpu manager policy is static
+	// 2. pod has quos PodQOSGuaranteed
+	// 3. container has integer cpu request
+	return cpuManagerPolicyName == string(PolicyStatic) &&
+		podQos == v1.PodQOSGuaranteed &&
+		cpuRequest.Value()*1000 == cpuRequest.MilliValue()
+}
+
```

Comment:
this seems to be used only in tests (and it's stateless, which is good). If so, let's move it into test code.
I'm not convinced we _need_ it for tests, but let's start moving it there.

Reply:
it is used within `calculateLinuxResources` to set resources.CpuQuota = int64(-1) at https://github.com/kubernetes/kubernetes/pull/127525/files/9037f77dada27c3fdcc6a26cda1db41cce1ade98#diff-f42b5c2632abe72391ae4fa91f7b51ebeb6266c7e10f6d6f472663cf97a493e8R249

instead of passing qos class into calculateLinuxResources we could compute that above and pass something like "cfsQuotaEnforced" to calculateLinuxResources to eliminate this